solr/test-framework/src/java/org/apache/solr/EmbeddedSolrServerTestBase.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.io.Writer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.lucene.util.LuceneTestCase;
+import org.apache.solr.client.solrj.embedded.EmbeddedSolrServer;
+import org.apache.solr.common.util.ContentStream;
+import org.apache.solr.common.util.ContentStreamBase;
+import org.apache.solr.common.util.ContentStreamBase.ByteArrayStream;
+import org.apache.solr.util.ExternalPaths;
+import org.junit.After;
+import org.junit.AfterClass;
+
+import com.google.common.io.ByteStreams;
+
+abstract public class EmbeddedSolrServerTestBase extends SolrTestCaseJ4 {
+
+  protected static final String DEFAULT_CORE_NAME = "collection1";
+
+  public static EmbeddedSolrServer client = null;
+
+  @After
+  public synchronized void afterClass() throws Exception {
+    if (client != null) client.close();
+    client = null;
+  }
+
+  @AfterClass
+  public static void afterEmbeddedSolrServerTestBase() throws Exception {
+
+  }
+
+  public synchronized EmbeddedSolrServer getSolrClient() {
+    if (client == null) {
+      client = createNewSolrClient();
+    }
+    return client;
+  }
+
+  /**
+   * Create a new solr client. Subclasses should override for other options.
+   */
+  public EmbeddedSolrServer createNewSolrClient() {
+    return new EmbeddedSolrServer(h.getCoreContainer(), DEFAULT_CORE_NAME) {
+      @Override
+      public void close() {
+        // do not close core container
```

Comment:
😄  see my ticket https://issues.apache.org/jira/browse/SOLR-15085 although ofc that's only in master and 8.x